### PR TITLE
add support to programmatically send the invitation email on expo

### PR DESCRIPTION
**Pull request by bbaayezi**

### CODE REVIEW

**Review by lukasotocerny — CHANGES_REQUESTED** [review]
It seems that there is lots of code duplication in `ExpoUtil` file, across functions `invite`, `cancelInvitation` and `login`. Could you consolidate the code, create some reusable method?

File: src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java [hunk 1]
```diff
@@ -64,4 +84,121 @@ public boolean publish(String projectID, String projectName, UUID conversionID)
...
+                                    .get("createUserInvitationForAccount").getAsJsonObject()
+                                    .get("id").getAsString();
+        } catch (IOException e) {
+            logger.error("Error sending invitation: " + e.fillInStackTrace());
+            return null;
+        }
+    }
+
+    public boolean cancelInvitation(String invitationID) throws IOException {
+        if (sessionString.equals("")) {
```
Comment (lukasotocerny) [inline src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java:162]: I'd make `cancelInvitation` consistent with `invite` function, so call here `login` as well (and within `login` check `sessionString` as is in my next comment)

File: src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java [hunk 2]
```diff
@@ -64,4 +84,121 @@ public boolean publish(String projectID, String projectName, UUID conversionID)
...
+        return false;
+    }
+
+    /**
+     * This method will login to the expo.io and get the session object string
+     * @param username
+     * @param password
+     * @return
+     */
+    public JsonObject login(String username, String password) throws IOException {
```
Comment (lukasotocerny) [inline src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java:187]: within `login` function I'd check if `sessionString` already exists (and potentially if it has valid `expiresAt` timestamp, but that is in my opinion redundant, as their sessionStrings expire at 10 years time), and if it does do not perform API call

**Review by lukasotocerny — CHANGES_REQUESTED** [review]

File: src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java (hunk 2 above)
Comment (bbaayezi) [reply src/main/java/com/ucl/imaginethisserver/Util/ExpoUtil.java:187]: Good point, I think I can send a request to clear all other sessions after successfully logged in.

**Comment by bbaayezi** [comment]
I Will keep this PR open for now, planning to add better error handling

**Comment by bbaayezi** [comment]
Merge it first for the MVP, will continue working on this branch

**Pull request merged**